### PR TITLE
athena: change dataset name to its location

### PR DESCRIPTION
**Pull request by sophiely**
### Problem

According to the discussion here: https://github.com/OpenLineage/OpenLineage/issues/435
we should name the dataset according to its location and not the catalog.
Which also make sense here because other jobs from other integration can also read the very same data from it's location and now with the current behavior we are not able to link them with airflow athena jobs.

So the lineage is incomplete and we have a duplicate of the same dataset with different name (one with the location from s3 and another within the athena namespace).

Closes: [2166](https://github.com/OpenLineage/OpenLineage/issues/2166)

### Solution

The location of the data can be retrieve in athena get-meta-data function so we can replace the dataset name and namespace with the data physical location. We can also add a symlinks facets with the name of the catalog, the database and the table in Athena (like the discussion about the hive metastore/physical location linked).

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Comment by pawel-big-lebowski** [comment]
@sophiely Please be aware, that the issue https://github.com/OpenLineage/OpenLineage/issues/435 you mentioned, has lead us to symlinks feature, which allows us registering dataset under multiple names. 

In case of spark integration, we decided to send physical location as a primary dataset identifier and include catalog identifier within the symlink facet. However, this approach is not obligatory. 

On the Marquez side, we implemented symlink feature in a way that, when a symlink for a dataset is sent, Marquez will know a dataset can be identified by different names. So, if you have a table in object storage that can be accessed by object storage physical location or by relational catalog on the top of it (like Athena), you just need to send a single OL event with the symlink facet. It does not matter if which idenfiier you put within dataset definition and which one will be contained in symlink facet. This will let Marquez know, that both identifiers correspond to the same dataset for all the other jobs, which can access data by object storage location or table name in catalog. 

To sum up: I think it's not required to replace existing identifier. Including a symlink should be sufficient.

**Comment by pawel-big-lebowski** [comment]
@JDarDagran wdyt?

**Comment by JDarDagran** [comment]
I think it's reasonable to do the same thing we do in Spark integration:

>  send physical location as a primary dataset identifier and include catalog identifier within the symlink facet.

This PR seems to do so.

**Comment by dkt-sophie-ly** [comment]
Hi all ! 

Thanks for your reply.
So I understand that we just need to send the physical location information of the data in the symlink facets.

For now marquez seems to have issue with symlinks so I can't really say if symlink facet will solve my [issue](https://github.com/MarquezProject/marquez/issues/2645) or not.

Besides I don't think that athena extractor include s3 location in a symlink facets for the moment, should we change the PR in this direction ?

Also, for a dataset read by Athena the dataSource is indeed awsathena://athena.{region_name}.[amazonaws.com](http://amazonaws.com/) but the actual source of the data is s3, like you said @pawel-big-lebowski athena is just a relational catalog on top of it. Don't you think it can be misleading to call it dataSource then ?

**Comment by JDarDagran** [comment]
> Besides I don't think that athena extractor include s3 location in a symlink facets for the moment, should we change the PR in this direction ?

You're right. We should add s3 location in a symlink facet.

> Also, for a dataset read by Athena the dataSource is indeed awsathena://athena.{region_name}.[amazonaws.com](http://amazonaws.com/) but the actual source of the data is s3, like you said @pawel-big-lebowski athena is just a relational catalog on top of it. Don't you think it can be misleading to call it dataSource then ?

That really depends. First (and I share it) perspective is that data comes from Athena and that is where dataSource is pointing at. We can also dig deeper and attach same dataset symlink to its actual blob location.

**Comment by dkt-sophie-ly** [comment]
Hi all ! Just changed the PR according to what we have discussed.
Add s3 location in symlink facet

**Comment by JDarDagran** [comment]
@dkt-sophie-ly thanks for the changes. We definitely should add some unit tests and mock some responses from API. This is a bigger task so I can take care of that in case it's too difficult.

**Comment by dkt-sophie-ly** [comment]
Hi @JDarDagran 

Just added some tests for athena extractor :) let me know if it's ok for you

File: integration/airflow/tests/extractors/test_athena_extractor.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    SymlinksDatasetFacet,
+    SymlinksDatasetFacetIdentifiers,
+)
+from openlineage.common.dataset import Dataset, Field, Source
+from openlineage.common.models import DbColumn, DbTableSchema
+from openlineage.common.sql import DbTableMeta
+
+from airflow import DAG
+
+if is_airflow_version_enough("2.2.4"):
```
Comment (JDarDagran) [inline integration/airflow/tests/extractors/test_athena_extractor.py:20]: Any reason this is 2.2.4 here but skips on 2.2.1?

**Review by JDarDagran — APPROVED** [review]
That's a first-class PR 🚀 ! Thank you for your contribution!
One small question though.

File: integration/airflow/tests/extractors/test_athena_extractor.py (hunk 1 above)
Comment (dkt-sophie-ly) [reply integration/airflow/tests/extractors/test_athena_extractor.py:20]: Hi @JDarDagran do you mean 2.1.4 ? 

Since AthenaHook is not compatible with airflow 2.1.4 I want to run these tests if airflow version >= 2.2.4

Tests for airflow 2.2.4:
image

Tests for airflow 2.1.4:
image

Comment (dkt-sophie-ly) [reply integration/airflow/tests/extractors/test_athena_extractor.py:20]: Oh just a typo on my side, i'll fix this right now.

Thanks for noticing it 


```

Comment (JDarDagran) [reply integration/airflow/tests/extractors/test_athena_extractor.py:20]: Thank you!
